test/storage.js

```diff
@@ -0,0 +1,111 @@
+'use strict'
+var tape = require('tape')
+var contractStorageResources = require('./resources/contractStorage')
+var stateDecoder = require('../src/solidity/stateDecoder')
+var solcWrapper = require('solc/wrapper')
+
+tape('Storage', function (t) {
+  console.log('loading last compiler version...')
+  solcWrapper().loadRemoteVersion('latest', function (err, compiler) {
+    if (err) {
+      t.fail(err)
+    } else {
+      test(t, compiler)
+    }
+  })
+})
+
+function test (t, solc) {
+  t.test('Storage.typedecoder.all', function (st) {
+    var output = solc.compile(contractStorageResources.testSimpleStorage, 0) // 1 activates the optimiser
+    var state = stateDecoder.getStateVariableLocations('testSimpleStorage', output.sources)
+
+    checkType(st, state[0].type, 'uint32', 4)
```

Comment:
I'm not sure if this is the  best way to test this. Perhaps it makes more sense to have more locally-constrained tests, where the source code and the storage locations are closer to each other in the code. For example you give a short contract: `contract c { uint16 a; uint64 b;}` and then check the storage locations of only those two variables.